examples/rdl.py

```diff
@@ -0,0 +1,710 @@
+see https://github.com/snap-stanford/relbench or
+https://github.com/kumo-ai/ContextGNN for link prediction.
+"""
+import argparse
+import copy
+import math
+import os
+from typing import Any, Dict, List, Optional
+
+import numpy as np
+import torch
+import torch_frame
+from relbench.base import EntityTask, TaskType
+from relbench.datasets import get_dataset
+from relbench.modeling.graph import (
+    get_node_train_table_input,
+    make_pkey_fkey_graph,
+)
+from relbench.modeling.utils import get_stype_proposal
+from relbench.tasks import get_task
+from sentence_transformers import SentenceTransformer
+from torch import Tensor
+from torch_frame.config.text_embedder import TextEmbedderConfig
+from torch_frame.data.stats import StatType
+from torch_frame.nn.models import ResNet
+from tqdm import tqdm
+
+from torch_geometric.data import HeteroData
+from torch_geometric.loader import NeighborLoader
+from torch_geometric.nn import (
+    MLP,
+    HeteroConv,
+    LayerNorm,
+    PositionalEncoding,
+    SAGEConv,
+)
+from torch_geometric.seed import seed_everything
+from torch_geometric.typing import EdgeType, NodeType
+
+seed_everything(42)
+
+device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+root_dir = "./data"
+print("Using device: ", device)
+
+args = argparse.ArgumentParser()
+
+# Dataset argument
+args.add_argument(
+    "--dataset", type=str, default="rel-f1", choices=[
+        "rel-stack", "rel-amazon", "rel-trial", "rel-f1", "rel-hm",
+        "rel-event", "rel-avito"
+    ])
+known_args, _ = args.parse_known_args()
+
+# task argument choices based are based on the dataset, this example only
+# supports node level tasks
+if known_args.dataset == "rel-f1":
+    args.add_argument("--task", type=str, default="driver-position",
+                      choices=["driver-position", "driver-dnf", "driver-top3"])
+elif known_args.dataset == "rel-amazon":
+    args.add_argument(
+        "--task", type=str, default="user-churn",
+        choices=["user-churn", "item-churn", "user-ltv", "item-ltv"])
+elif known_args.dataset == "rel-hm":
+    args.add_argument("--task", type=str, default="user-churn",
+                      choices=["user-churn", "item-sales"])
+elif known_args.dataset == "rel-stack":
+    args.add_argument("--task", type=str, default="user-engagement",
+                      choices=["user-engagement", "user-badge", "post-votes"])
+elif known_args.dataset == "rel-trial":
+    args.add_argument(
+        "--task", type=str, default="study-outcome",
+        choices=["study-outcome", "study-adverse", "site-success"])
+elif known_args.dataset == "rel-event":
+    args.add_argument(
+        "--task", type=str, default="user-repeat",
+        choices=["user-repeat", "user-ignore", "user-attendance"])
+elif known_args.dataset == "rel-avito":
+    args.add_argument("--task", type=str, default="user-visits",
+                      choices=["user-visits", "user-clicks", "ad-ctr"])
```

Comment:
This is both very clean and very rigor at the same time. IMO, we can just do `args.add_argument('--dataset')` and `args.add_argument('--task')`. We will fail later anyway on a mismatch between the two.